webapi/src/main/scala/org/knora/webapi/routing/RouteUtilZ.scala

```diff
@@ -0,0 +1,31 @@
+package org.knora.webapi.routing
+import zio.Task
+import zio.ZIO
+
+import java.net.URLDecoder
+
+import dsp.errors.BadRequestException
+
+object RouteUtilZ {
+
+  /**
+   * Url decode a [[String]].
+   * Fails if String is not a well formed utf-8 [[String]] in `application/x-www-form-urlencoded` MIME format.
+   *
+   * Wraps Java's [[java.net.URLDecoder#decode(java.lang.String, java.lang.String)]] into the zio world.
+   *
+   * @param value The value in utf-8 to be url decoded
+   * @param errorMsg Custom error message for the error type
+   * @return ```success``` the decoded string
+   *
+   *         ```failure``` A [[BadRequestException]] with the `errorMsg`
+   */
+  def urlDecode(value: String, errorMsg: String = ""): Task[String] =
```

Comment:
Do we actually need the second parameter `errorMsg`?

Reply:
Yes, different Routes might want create their own error message. In this PR it is "Failed to url decode IRI parameter." in the route. I did not want to hardcode that particular message into the RouteUtil as it should not assume that all encoded params are IRIs.